recipes/grpc/all/conanfile.py

```diff
@@ -92,7 +92,7 @@ def requirements(self):
         else:
             self.requires("abseil/20220623.0", transitive_headers=True)
         self.requires("c-ares/1.18.1")
-        self.requires("openssl/1.1.1t")
+        self.requires("openssl/[>=1.1 <4]")
```

Comment:
I believe that conancenter should first clarify policies regarding version range:
- for conan v1 & v2?
- upper bound or not (if unknown like here)?

Reply:
I'm currently working on updating the documentation for this.

For the time being, we will only allow OpenSSL and CMake (as a `tool_requires`), as these have very well known compatibility paths and version strategies (unlikely to break on a minor or patch). We will monitor how well this works for Conan Center (both users and our CI services) before making any further decisions or allowing it more generally.

Reply:
Ok, it's a good one to start with.

Once stabilized, other widely used libraries with quite stable API may benefit from version range in downstream recipes:

- zlib (140 recipes)
- libjpeg (42 recipes)
- libpng (40 recipes)
- libcurl (42 recipes)
- libiconv (34 recipes)
- zstd (29 recipes)
- sqlite3 (26 recipes)
- libxml2 (26 recipes)
- libtiff (25 recipes)
- freetype (23 recipes)
- xz_utils (20 recipes)
- bzip2 (19 recipes)
- lz4 (16 recipes)
- icu (15 recipes)
- libwebp (15 recipes)